### PR TITLE
Gregungerer/arm exidx fix

### PR DESCRIPTION
**Pull request by gregungerer**
This is a different approach to fixing the issues that elf2flt has had with dumping and invalid relocations that stem from the ARM.exidx section problem. This reverts the fix and follow up fix that still has breakage on some architectures.

The change here now is to explicitly include the ARM.exidx section in the elf2flt text section in the linker script - instead of keeping it as a separate named section. With that section now inside the text section proper there is no sizing or alignment problems for and between the text and data sections.

This approach should also means that no further special cases are required fix eh_frames and similar issues.

### CODE REVIEW

File: elf2flt.c [hunk 1]
```diff
@@ -372,7 +372,7 @@ output_relocs (
 #endif
 
 #if 0
-  printf("%s(%d): output_relocs(abs_bfd=%d,symbols=0x%x,number_of_symbols=%d,"
+  printf("%s(%d): output_relocs(abs_bfd=%d,synbols=0x%x,number_of_symbols=%d"
```
Comment (vapier) [inline elf2flt.c]: unfortunately it seems that we had some unrelated fixes in the original changes :(.  can we preserve these ?  maybe fix & squash into the reverts so the reverts are just partial ?

File: elf2flt.c [hunk 2]
```diff
@@ -428,13 +428,10 @@ output_relocs (
 //		continue;
 
 	/*
-	 * Only relocate things in the writable data sections if we are PIC/GOT.
-	 * Otherwise do text (and read only data) as well.
+	 *	Only relocate things in the data sections if we are PIC/GOT.
```
Comment (vapier) [inline elf2flt.c]: same here

File: elf2flt.c (hunk 1 above)
Comment (gregungerer) [reply elf2flt.c]: I can certainly do that. A very good reason to not mix unrelated changes in a single commit.

**Review by vapier — APPROVED** [review]
